### PR TITLE
OCM-5285: MP labels map must have a size of at least 1

### PR DESCRIPTION
**Pull request by sagidayan**
If users wish to delete all labels, they should simply remove the `labels` attribute from the `tf` statement.

### CODE REVIEW

File: subsystem/machine_pool_resource_test.go [hunk 1]
```diff
@@ -662,11 +662,23 @@ var _ = Describe("Machine pool creation", func() {
...
+		  resource "rhcs_machine_pool" "my_pool" {
+		    cluster      = "123"
+		    name         = "my-pool"
+		    machine_type = "r5.xlarge"
+		    replicas     = 12
+            labels       = {}
+		  }
+		`)
+		Expect(terraform.Apply()).ToNot(BeZero())
+		// Valid deletion
```
Comment (bardielle) [inline subsystem/machine_pool_resource_test.go:681]: Please also add an update flow:
1. Create MP with labels 
2. remove them by setting it to empty map -> should fail 
3. Remove them by setting it to null -> should succeed

**Comment by bardielle** [comment]
@sagidayan please change the base branch to `v1.5.0-branch` and cherry pic to main

File: subsystem/machine_pool_resource_test.go (hunk 1 above)
Comment (sagidayan) [reply subsystem/machine_pool_resource_test.go:681]: That is what I am doing in the test. 🤔  

in tf `=null` is like not passing the attribute. I can add that case but seems pointless. wdyt?

**Comment by sagidayan** [comment]
PR was messed up. Will create a new one.

**Pull request closed without merging**